src/main/java/org/cirdles/topsoil/app/CloseTabEventHandler.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2015 CIRDLES.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.cirdles.topsoil.app;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+import javafx.event.Event;
+import javafx.event.EventHandler;
+import javafx.scene.control.Alert;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.Tab;
+import org.cirdles.topsoil.dataset.DatasetResource;
+import org.cirdles.topsoil.dataset.DatasetResourceManager;
+
+/**
+ *
+ * @author parizotclement
+ */
+public class CloseTabEventHandler implements EventHandler<Event> {
+
+    private final Tab tab;
+    private final DatasetResourceManager datasetResourceManager;
+    private final Consumer<Boolean> saveDataTable;
+
+    public CloseTabEventHandler(Tab tab, DatasetResourceManager datasetResourceManager, Consumer<Boolean> callback) {
```

Comment:
The callback method is in fact the `saveDataTable()` method from `TopsoilMainWindow`. It is already used elsewhere, won't it be recurrent to redefine the exact same method in another class instead of just calling it ?